### ISSUE TITLE
Make the extension compatible with VS Code Web

### ISSUE THREAD

**Issue by AndreaPi** [yzhang-gh/vscode-markdown#996]
## Proposal

Make the extension compatible with VS Code Web. Currently it isn't.

## References

![image](https://user-images.githubusercontent.com/8233615/129456279-b1831869-d093-470e-ba52-42d006b62fdd.png)

**Comment by Lemmingh** [comment]
Duplicate of #948

**Comment by yzhang-gh** [comment]
I've enabled the virtual workspaces support although some features may not work as expected.

People who are interested can install the [dev build](https://github.com/yzhang-gh/vscode-markdown#latest-development-build).
https://code.visualstudio.com/api/advanced-topics/remote-extensions#installing-a-development-version-of-your-extension

**Comment by Lemmingh** [comment]
Too many modules depend on sync `fs`, while VS Code's `fs` is async. That's the major problem.

**Comment by Lemmingh** [comment]
According to a [recently](https://code.visualstudio.com/updates/v1_60) published [authoring guide](https://code.visualstudio.com/api/extension-guides/web-extensions), there is much more to do than just supporting Virtual Workspaces.

**Comment by isidorn** [comment]
Hi, VS Code PM here 👋 

Yeah, we would love to have `vscode-markdown` available on the web if that is possible. 
And we have a guide for enabling extensions for the web [here](https://code.visualstudio.com/api/extension-guides/web-extensions). As a couple of highlights:

- In VS Code for the Web, both the UI and extension host run inside the browser.
- A web extension is structured like a regular extension, but with a different main file: it's defined by the `browser` property
- Access to workspace files needs to go through the VS Code file system API accessible at `vscode.workspace.fs`
- There are currently [three ways](https://code.visualstudio.com/api/extension-guides/web-extensions#test-your-web-extension) to test a web extension

@yzhang-gh  Please let us know if you have questions or feedback on enabling your extension for the web - our team would love to learn from your experiences and collaborate. Thank you!

**Comment by yzhang-gh** [comment]
Thanks for reaching out.

I am willing to make this extension available for the VS Code Web. However, I won't have enough time to do this until (exactly) one month later...

**Comment by isidorn** [comment]
This is great news. In a couple of months would be great!

**Comment by 0x-jerry** [comment]
I have published a fork of this extension that supports vscode web with some limitations. [markdown-all-in-one-for-web](https://marketplace.visualstudio.com/items?itemName=fantasy.markdown-all-in-one-for-web)

Limitations:

- No NLS, only English.
- No print to pdf.
- No welcome message.

I wish this could help you.

Because of the limitations, it can not merge directly, so I am not making a PR. Also, I am not fully testing the change I made.

Btw, Thank you made the great extension.  @yzhang-gh

**Comment by isidorn** [comment]
@0x-jerry thanks for working on this. Instead of publishing a fork can you instead open a PR to this extension. I think that would be more helpful? Thanks a lot!

**Comment by 0x-jerry** [comment]
@isidorn Yes, I agree with you. But as I mentioned above, there have some limitations that need to be solved. And it needs time to solve that which I don't have recently. That is why I am not open a PR, I will try to solve those limitations and open a PR when I have time, but that may take a long time.

**Comment by Lemmingh** [comment]
Prerequisites:

* Build system: #1054

* NLS: #1025

* POSIX path utilities: <https://github.com/nodejs/node/blob/master/lib/path.js>

* Base64 Encoding API: <https://github.com/nodejs/node/blob/master/src/base64-inl.h>

**Comment by yzhang-gh** [comment]
Unfortunately no as it requires a relative long period of time to sort all these things out.
